### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by PaluMacil** [dop251/goja#76]
I'd like to start contributing to Annex B and then some ES 2015 functionality. To do that I think it would be beneficial to list features in general order of priority. I don't mind doing a lot of legwork to put together a document if you're up for reviewing my doc PR and telling me how I am or am not following your vision for this project. Would you be interested in that sort of contribution?

**Comment by dop251** [comment]
I did consider implementing ES2015 but decided it would provide too little benefit. ES5.1 is de-facto the "assembly" language of the Javascript ecosystem. There are transpilers from many languages and dialects (including ES2015) into ES5.1. And it will remain the case at least for as long as IE is around which will be a very long time.

Having only _partial_ ES2015 support has limited value. It's ok if you write code specifically for goja so you know which features to use and which to avoid, but what if you need to use a 3rd part library? You'd have to use a transpiler anyway so why doing its job? There are a few exceptions like Proxy which is much easier to do with runtime support, so that I would like to be added. I know @noctarius is working on it and I look forward to merging it when it's ready.

Bottom line, if you feel like contributing please do, I will consider any PRs that are good quality code, pass the relevant tc39 tests, don't break existing stuff and don't degrade performance. However unless you are prepared to do full ES2016 implementation (which is a huge undertaking) I would not waste time.

**Comment by PaluMacil** [comment]
Though you mention proxy, all the major browsers support let, const, and string templates. Would those be exceptions in your mind since they are common in implementations that are basically 5.1+?
